src/test/java/LinkedIteratorTest.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class LinkedIteratorTest {
+
+    @Test
+    void hasNext_emptyLinked_returnsFalse() {
```

Comment:
emptyLinked -> emptyList ? It's not really clear what you mean by _linked_ here. It sounds more like a single element list, with `null` linked to the end.  
_Linked_ terminology occurs throughout.